src/contracts/libraries/SlashingLib.sol

```diff
@@ -0,0 +1,181 @@
+ *          - These live in the storage of the DelegationManager:
+ *              - `delegatedShares` is the total amount of delegatedShares delegated to an operator for a strategy
+ *              - `withdrawal.delegatedShares` is the amount of delegatedShares in a withdrawal          
+ *      3. shares 
+ *          - These can be converted into delegatedShares given a staker and a strategy
+ *              - by multiplying by the staker's depositScalingFactor for the strategy
+ *          - These values automatically update their conversion into tokens
+ *             - when the staker's depositScalingFactor for the strategy is increased upon new deposits
+ *             - or when the staker's operator's total magnitude for the strategy is decreased upon slashing
+ *          - These represent the total amount of shares the staker would have of a strategy if they were never slashed
+ *          - These live in the storage of the StrategyManager/EigenPodManager
+ *              - `stakerStrategyShares` in the SM is the staker's shares that have not been queued for withdrawal in a strategy
+ *              - `podOwnerShares` in the EPM is the staker's shares that have not been queued for withdrawal in the beaconChainETHStrategy
+ */
+
+type OwnedShares is uint256;
+type DelegatedShares is uint256;
+type Shares is uint256;
+
+using SlashingLib for OwnedShares global;
+using SlashingLib for DelegatedShares global;
+using SlashingLib for Shares global;
+
+library SlashingLib {
+    using Math for uint256;
+    using SlashingLib for uint256;
+
+    function toShares(
+        DelegatedShares delegatedShares,
+        uint256 depositScalingFactor
+    ) internal pure returns (Shares) {
+        if (depositScalingFactor == 0) {
+            depositScalingFactor = WAD;
+        }
+
+        // forgefmt: disable-next-item
+        return delegatedShares
+            .unwrap()
+            .divWad(depositScalingFactor)
+            .wrapShares();
+    }
+
+    function toDelegatedShares(
+        Shares shares,
+        uint256 depositScalingFactor
+    ) internal pure returns (DelegatedShares) {
+        if (depositScalingFactor == 0) {
+            depositScalingFactor = WAD;
+        }
+
+        // forgefmt: disable-next-item
+        return shares
+            .unwrap()
+            .mulWad(depositScalingFactor)
+            .wrapDelegated();
+    }
+
+    function toOwnedShares(
+        DelegatedShares delegatedShares, 
+        uint256 magnitude
+    ) internal pure returns (OwnedShares) {
+        // forgefmt: disable-next-item
+        return delegatedShares
+            .unwrap()
+            .mulWad(magnitude)
+            .wrapOwned();
+    }
+
+    function toDelegatedShares(
+        OwnedShares shares, 
+        uint256 magnitude
+    ) internal pure returns (DelegatedShares) {
+        // forgefmt: disable-next-item
+        return shares
+            .unwrap()
+            .divWad(magnitude)
+            .wrapDelegated();
```

Comment:
This is getting a bit cumbersome :grimacing: 

4 "functions" to represent some pretty simple math

Reply:
Could add more mulWad and divWad functions that take in the same type rather than always returning a uint256 to reduce the need for unwrap and wrap during calculations that don't live in the lib itself.